webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/repository/ldap/LDAPNodeGroupRepository.scala

```diff
@@ -579,13 +581,15 @@ class WoLDAPNodeGroupRepository(
     groupLibMutex.readLock(
       for {
         con              <- ldap
-        oldCategoryEntry <- getCategoryEntry(con, category.id, "1.1").notOptional(s"Entry with ID '${category.id.value}' was not found")
-        categoryEntry    =  mapper.nodeGroupCategory2ldap(category,oldCategoryEntry.dn.getParent)
-        exists           <- categoryExists(con, category.name, oldCategoryEntry.dn.getParent, category.id)
-        canAddByName     <- if (exists)
-                              s"Cannot update the Node Group Category with name '${category.name}': a category with the same name exists at the same level".fail
-                            else UIO.unit
-        result           <- groupLibMutex.writeLock { con.save(categoryEntry, removeMissingAttributes = true) }
+        result           <- groupLibMutex.writeLock { for {
```

Comment:
i think the method should be in a riteLock and not a readLock

Reply:
yes